rust_src/src/editfns.rs

```diff
@@ -297,6 +298,24 @@ pub fn following_char() -> EmacsInt {
     }
 }
 
+/// Return the character preceding point, as a number. At the
+/// beginning of the buffer or accessible region, return 0.
+#[lisp_fn(c_name = "previous_char")]
+pub fn preceding_char() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+
+    if buffer_ref.pt <= buffer_ref.begv {
+        return LispObject::from_natnum(0);
```

Comment:
Since you are returning `natnum` this routine can return `EmacsInt` instead of `LispObject`.